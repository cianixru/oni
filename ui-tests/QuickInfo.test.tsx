import { mount, shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"
import * as React from "react"

import * as os from "os"

import {
    QuickInfoDocumentation,
    QuickInfoTitle,
    Title,
} from "./../browser/src/UI/components/QuickInfo"

describe("<QuickInfo />", () => {
    it("renders a shallow instance of the component", () => {
        const wrapper = shallow(<QuickInfoTitle />)
        expect(shallowToJson(wrapper)).toMatchSnapshot()
        expect(wrapper.length).toEqual(1)
    })

    it("renders the correct text", () => {
        const wrapper = mount(<QuickInfoTitle text="test text" />)
        expect(wrapper.text()).toContain("test text")
    })

    it("renders the documentation correctly", () => {
        const test = ["line One", "line Two", "line Three"].join(os.EOL)
        const wrapper = shallow(<QuickInfoDocumentation text={test} />)
        expect(wrapper.children()).toHaveLength(3)
    })
})
