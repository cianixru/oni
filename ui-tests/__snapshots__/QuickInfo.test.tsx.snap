// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuickInfo /> renders a shallow instance of the component 1`] = `""`;
